### ISSUE TITLE
trove_arbitrum.trades is not indexing all smart contracts

### ISSUE THREAD

**Issue by 0xBoxer** [duneanalytics/spellbook#3170]
### Description
trove_arbitrum.trades is missing data between march 2022 and june 2022.

A query with the complete data:
https://dune.com/queries/2276649/3731171

Our lacking trove_arbitrum.trades data 
https://dune.com/queries/2363756

### Current behavior
Probably some smart contracts are missing, hence there is a gap in the reported data

### Expected behavior
All smart contracts are part of the spell

### Impacted model(s)
trove_arbitrum.trades
trove.trades

### Possible solution
Add the missing smart contract data into the Spell

**Comment by jeff-dude** [comment]
have any helpful wizards to tag to help us add?

**Comment by 0xBoxer** [comment]
I think @bizzyvinci initially added trove on Arbitrum, but the solution is already in the queries above, anyone should be able to pick this up.

**Comment by bizzyvinci** [comment]
I've figured out the issue, and help would be required from decoding.

TLDR: ABI of ItemSold changed within v2 and proposed solution is at the end

The missing contract for march to june 2022 is v2. [v1](https://arbiscan.io/address/0x2e3b85f85628301a0bce300dee3a6b04195a15ee) doesn't have successful sale since march 2022, which was when [v2](https://arbiscan.io/address/0x09986b4e255b3c548041a30a2ee312fe176731c2) was [created](https://arbiscan.io/tx/0x43b348d6e0eee06b25bb184f28399c82f8c9793e6f55c6f80754448dd0a945f5)

In the spells, v1 was added using `ItemSold` decoded events, while v2 was added using `ItemSold` and `BidAccepted` events.

[this query](https://dune.com/queries/2276649/3731171) used raw topics to index v2.
* L19 0x8f236686c9ab7948a1cc5985ed36b4de9780519f75380d5ea3ac498f9270e17f
* L21 0x72d3f914473a393354e6fcd9c3cb7d2eee53924b9b856f9da274e024566292a5

We can search for events [here](https://arbiscan.io/address/0x09986b4e255b3c548041a30a2ee312fe176731c2#events)
* `0x8f236686c9ab7948a1cc5985ed36b4de9780519f75380d5ea3ac498f9270e17f` is ItemSold (address seller, address buyer, address nftAddress, uint256 tokenId, uint64 quantity, uint128 pricePerItem) with the last transaction on `jun-13-2022`
* `0x72d3f914473a393354e6fcd9c3cb7d2eee53924b9b856f9da274e024566292a5` is ItemSold (address seller, address buyer, address nftAddress, uint256 tokenId, uint64 quantity, uint128 pricePerItem, address paymentToken) with transactions minutes or seconds ago

The difference is the last argument, `address paymentToken` which signifies the payment token, rather than the default [Magic token](https://arbiscan.io/token/0x539bde0d7dbd336b79148aa742883198bbf60342). V2 is a proxy contract and according to arbiscan, the current implementation is [0x3204192f4d87fbaf87a4c13d1a1ad8ca04c478d7](https://arbiscan.io/address/0x3204192f4d87fbaf87a4c13d1a1ad8ca04c478d7#code) while the former implementation is [0x338faebf597a988f483b7964b287f02495a1b56e](https://arbiscan.io/address/0x338faebf597a988f483b7964b287f02495a1b56e#code). However, both of them has paymentToken in ItemSold event.

There are 2 potential solutions
* Use raw data to index data
* Find the other implementation that emits 0x8f236686c9ab7948a1cc5985ed36b4de9780519f75380d5ea3ac498f9270e17f and add them to decoded data. It might not be easy especially if the ABI is significantly different. But for ItemSold, paymentToken could be null or changed to Magic

**Comment by bizzyvinci** [comment]
@0xBoxer @jeff-dude I was able to trace [0xcb9f3d6a7c4a8eee1ae8668c05b860cbc0a6dbc3](https://arbiscan.io/address/0xcb9f3d6a7c4a8eee1ae8668c05b860cbc0a6dbc3) as one of the implementation that's missing ([proof](https://dashboard.tenderly.co/tx/arbitrum/0x5e39f062be60252cd0fca58d0ffcaab61532956c699923892ce62b2fb586f0a1/debugger?trace=0.0)). There might be another implementation cause it was deployed on mar-29 while the last item sold on v1 was on mar-03 while.

**Comment by bizzyvinci** [comment]
Yes, please.
This [query](https://dune.com/queries/2407113) shows all the implementation contracts. So far, I know there's a difference in ItemSold event and there might be more. But I believe the decode team would have a way to merge them.

**Comment by jeff-dude** [comment]
@antonio-mendes curious if you can help look into this issue around decoded contracts for trove on arbitrum?

**Comment by bizzyvinci** [comment]
Thanks all. #3245 should resolve the issue.

**Comment by bizzyvinci** [comment]
Also, a lot of usd volume is missing cause MAGIC **token address** is not on coinpaprika.
The token is actually available on [coinpaprika](https://coinpaprika.com/coin/magic-magic-arbitrum/) because the price chart looks similar to that of [coinmarketcap](https://coinmarketcap.com/currencies/magic-token/).
You might need to contact their support to update the contract address and **logo**.

**Comment by jeff-dude** [comment]
> Edit: I was wrong, ignore below message. Their price is included but didn't start until may-9.
> 
> Also, a lot of usd volume is missing cause MAGIC **token address** is not on coinpaprika. The token is actually available on [coinpaprika](https://coinpaprika.com/coin/magic-magic-arbitrum/) because the price chart looks similar to that of [coinmarketcap](https://coinmarketcap.com/currencies/magic-token/). You might need to contact their support to update the contract address and **logo**.

not 100% sure i follow. from what i see, the address/symbol are correct in spellbook:
https://github.com/duneanalytics/spellbook/blob/main/models/prices/arbitrum/prices_arbitrum_tokens.sql#L59

then the coinpaprika api returns as expected:
https://api.coinpaprika.com/v1/coins/magic-magic-arbitrum

with links to arbiscan:
https://arbiscan.io/token/0x539bdE0d7Dbd336b79148AA742883198BBF60342

**Comment by bizzyvinci** [comment]
Yes, you are correct @jeff-dude

**Comment by jeff-dude** [comment]
got it -- so we aren't missing trades at this point (if the attached PR is merged), but `amount_usd` may be missing due to coinpaprika gaps? so we should proceed with the PR?

**Comment by bizzyvinci** [comment]
Yes
